src/spikeinterface/core/generate.py

```diff
@@ -801,11 +821,84 @@ def synthesize_random_firings(
     times = np.concatenate(times)
     labels = np.concatenate(labels)
 
-    sort_inds = np.argsort(times)
-    times = times[sort_inds]
-    labels = labels[sort_inds]
+    train_length = len(times)
+
+    spike_train = np.zeros(train_length, dtype=minimum_spike_dtype)
+    spike_train["sample_index"] = times
+    spike_train["unit_index"] = labels
+
+    spike_train = add_insertions(spike_train, insertions, insertions_replace, rng)
+
+    spike_train.sort(order=["sample_index"])
+
+    segment_length = train_length // segments
+    for i in range(train_length):
+        spike_train["segment_index"][i] = min(i // segment_length, segments - 1)
+
+    return spike_train
+
+
+def add_insertions(spike_train, insertions, insertions_replace, rng=None, seed=None):
+    """
+    Add specified insertions into a list of times and unit labels.
+
+    Parameters
+    ----------
+    spike_train: np.ndarray
+        Structured numpy array ("sample_index", "unit_index", "segment_index").
+    insertions: array-like
+        List of manually inserted spikes, each spike formatted as [sample index, unit index]. Intended for
+        use in testing.
+    insertions_replace: bool
+        If True, randomly replace generated spikes. If False, add to generated spikes.
+    rng: numpy.random.Generator
+        A random number generator
+    seed: int, default: None
+        seed for the generator
+
+    Returns
+    -------
+    spike_train: np.ndarray
+        Structured numpy array ("sample_index", "unit_index", "segment_index").
+        Including insertions.
+
+    """
+
+    if insertions is None:
+        return spike_train
+
+    new_spike_train = spike_train
+
+    if rng is None:
+        rng = np.random.default_rng(seed=seed)
+
+    insertions = get_structured_insertions(insertions)
+    if insertions_replace:
+        replacement_indices = rng.choice(len(spike_train), len(insertions), replace=False)
+        new_spike_train[replacement_indices] = insertions
+    else:
+        new_spike_train = np.append(spike_train, insertions)
+
+    return new_spike_train
+
+
+def get_structured_insertions(insertions):
+
+    structured_insertions = np.zeros(
+        len(insertions), dtype=[("sample_index", "int64"), ("unit_index", "int64"), ("segment_index", "int64")]
+    )
+
+    insertions = np.array(insertions)
+    for i, insert in enumerate(insertions):
+
+        structured_insertions["sample_index"][i] = insert[0]
+        structured_insertions["unit_index"][i] = insert[1]
+        if len(insert) == 3:
+            structured_insertions["segment_index"][i] = insert[2]
+        else:
+            structured_insertions["segment_index"][i] = 0
 
-    return (times, labels)
+    return structured_insertions
```

Comment:
Same coment here about the output. Input and output are your contracts with your users. We should be careful with changing them (and careful to define them in the first place)

Reply:
Indeed, I'll add some documentation. I think this (and the add_insertions) functions don't have much use outside of testing. Should I "hide" them (i.e. make them _add_insertions etc)?

Reply:
If you think that they will only be used for internal testing then the best is to have them in the corresponding testing module I feel.

I think hiding by default is a good option and then you make it public once you feel confident about the interface and purpose. That minimizes the risk if you want to change the function later. But this is my opinion, I don't think that anyone else would find any problems if you want to add some functions.

Reply:
I'd prefer to keep them close to the synthesize_stuff functions, since they are closely related. I'm a fan of the "related code should be close together" mindset. Maybe they should all be in the testing module? But maybe for another time...

Great, I've hidden it and also added some documentation. Best of all worlds.

Reply:
I think this is a good decision. If we ever use the function more and we are decided on the signature, then we can make it public.